src/python/pants/backend/codegen/protobuf/go/rules.py

```diff
@@ -250,11 +251,20 @@ async def setup_full_package_build_request(
     )
     if result.exit_code != 0:
         raise ValueError(
-            f"Failed to analyze Go sources generated from {request.import_path}.\n\n"
-            f"stdout:\n{result.stdout.decode()}\n\n"
-            f"stderr:\n{result.stderr.decode()}\n\n"
-            "This may be a bug in Pants. Please report this issue at "
-            "https://github.com/pantsbuild/pants/issues/new/choose"
+            softwrap(
+                f"""
+                Failed to analyze Go sources generated from {request.import_path}.
+
+                stdout:
+                    {result.stdout.decode()}
+
+                stderr:
+                    {result.stderr.decode()}
```

Comment:
Also, I think it becomes slightly more readable to indent after the headings here. However, question is if we ought to `hard_wrap()` the decoded `stderr` and `stdout` to preserve the indentation in case of multiline output.

Reply:
Yeah, might be worth a helper to drive consistency. :thinking:

Reply:
That's a good point with not changing the original std{out,err}. I think we've used this pattern before:

```python
softwrap("blah blah") + "\n\n" + stdout + softwrap("blah blah")
```

Trying to put it in all in the same `dedent()` or `softwrap()` call messes things up by trying to change `stdout`.

Reply:
I see options:

- Exit `softwrap` for this (but hope the user spcified newlines right
- Introduce a help to indent after newlines we can use e.g. `{indent(4, result.stdout.decode()}`
- Use `%s` and make `softwrap` take varargs and do string substitution in a way that "just works"

Reply:
I'm partial to the first one, presuming that's my suggestion of using `softwrap` and `+` for the rest. It seems like the simplest approach for now.

If we find the `+` is too bothersome and this is happening frequently enough, then try to get `softwrap` to handle it all.

Reply:
My two cents is that the old way isn't "broken", it's only annoying to type. What matters more than anything is the end-user message is formatted correctly. If we can use `softwrap()` to make 98% of those cases more ergonomic for us, awesome. Not a huge deal if there are some cases it can't do perfectly (as long as we adapt so that users don't see a regression).

Reply:
+1 for defer a fix for this in a dedicated PR.

Reply:
> My two cents is that the old way isn't "broken", it's only annoying to type. What matters more than anything is the end-user message is formatted correctly.

I'd like to add that the more difficult it is to type/review the more likely we are to get the formatting wrong for the user. So the two are related.

---

Let's punt though. Incremental improvement is still improvement.

Reply:
reverted